src/Modules/CrestApps.OrchardCore.AI/Drivers/AIProfileDisplayDriver.cs

```diff
@@ -122,8 +122,31 @@ public override IDisplayResult Edit(AIProfile profile, BuildEditorContext contex
             model.TopP = context.IsNew ? _defaultAIOptions.TopP : metadata.TopP;
             model.UseCaching = metadata.UseCaching;
             model.AllowCaching = _defaultAIOptions.EnableDistributedCaching;
+            model.UseMicrophone = metadata.UseMicrophone;
+            model.SpeechToTextConnectionName = metadata.SpeechToTextConnectionName;
 
             model.IsSystemMessageLocked = profile.GetSettings<AIProfileSettings>().LockSystemMessage;
+
+            // Populate speech-to-text connections for the current provider
+            if (!_aiOptions.ProfileSources.TryGetValue(profile.Source, out var profileSource))
+            {
+                model.SpeechToTextConnections = [];
+            }
+            else if (_connectionOptions.Providers.TryGetValue(profileSource.ProviderName, out var provider))
+            {
+                model.SpeechToTextConnections = provider.Connections
+                    .Where(x => x.Value.TryGetValue("Types", out var types) && 
```

Comment:
Why using reflection here? We should be able to filter by connection Type property